Add ability to send CV to the lever api

mobinzk opened the pull request:
Changed the post method to use `multipart/form-data` as the api only accepts files via `multipart/form-data`.

mobinzk committed:
Add ability to send CV to the lever api

Changed the post method to use `multipart/form-data` as the api only accepts files via `multipart/form-data`.

peterkeating commented on ViewModels/LeverPostingApplyViewModel.cs:
```diff
@@ -59,6 +62,75 @@ public string ToJson
             }
         }
 
+        public MultipartFormDataContent ToDataFrom()
```
Rename this to `ToFormData`.

peterkeating replied:
SoundCloud isn't a fan of the length of this method (it's double what it recommends). Happy to chat through ideas on how we can reduce the length of it.

peterkeating commented on ViewModels/LeverPostingApplyViewModel.cs:
```diff
@@ -14,6 +16,7 @@ public class LeverPostingApplyViewModel
         public string Comments { get; set; }
         public Dictionary<string, string> Consent { get; set; }
         public string Email { get; set; }
+        public IFormFile Resume { get; set; }
```
Reposition to respect alphabetical ordering of properties.

peterkeating commented on ViewModels/LeverPostingApplyViewModel.cs:
```diff
@@ -59,6 +62,75 @@ public string ToJson
+            {
+                // Double qoutes are required for keys as the Lever API doesn't accept the key without double qoutes.
+                // Solution from: https://stackoverflow.com/a/21874413
+                { new StringContent(!string.IsNullOrEmpty(Comments) ? Comments : string.Empty), @"""comments""" },
```
I'd suggest moving the double quotes to it's own method (e.g. `FormatForLever`) as it's being used everywhere.

peterkeating commented on ViewModels/LeverPostingApplyViewModel.cs:
```diff
@@ -59,6 +62,75 @@ public string ToJson
+
+            if (Urls != null && Urls.Any())
+            {
+                foreach (var item in Urls)
```
You can shorten this by changing it to `Urls.Where(x => x.Value != null)`.

mobinzk committed:
Update and improvements to methods

- Update method names
- Order getters alphabetically
- Refactor ToFromData method

peterkeating commented:
@mobinzk You requested a review but I don't see any changes.

mobinzk commented:
> @mobinzk You requested a review but I don't see any changes.

I've pushed up my changes but they don't come here!

mobinzk committed:
Fix typo for FormData

peterkeating committed:
Refactor view model methods to extension method

mobinzk reviewed (commented):
Perfect, thank you.
